### PR TITLE
Reduce examples benchmark boilerplate

### PR DESCRIPTION
**Pull request by robertbastian**
This makes our examples easier to read.

### CODE REVIEW

**Review by sffc — CHANGES_REQUESTED** [review]
I very much dislike macros that wrap code because it breaks intellisense and sometimes formatting. I think the old boilerplate, while not perfect, is fine. I'm not convinced that this PR is an improvement, but I can be convinced.

**Comment by robertbastian** [comment]
ok

File: utils/zerotrie/examples/first_weekday_for_region.rs [hunk 1]
```diff
@@ -211,17 +210,10 @@ fn black_box<T>(dummy: T) -> T {
...
     // Un-comment to re-generate the bytes (printed to the terminal)
     // let trie_phf = DATA.iter().copied().collect::<ZeroTriePerfectHash<Vec<_>>>();
     // assert_eq!(trie_phf.as_bytes(), TRIE_PHF.as_bytes());
 
-    if black_box(TRIE_PHF).get(b"MV") == Some(weekday::FRI) {
-        0
-    } else {
-        1
-    }
+    assert_eq!(black_box(TRIE_PHF).get(b"MV"), Some(weekday::FRI));
```
Comment (sffc) [inline utils/zerotrie/examples/first_weekday_for_region.rs:218]: Nit: This probably pulls in the std formatter. Maybe just use `assert!`

**Review by sffc — DISMISSED** [review]
Much better, thanks!

File: utils/zerotrie/examples/first_weekday_for_region.rs (hunk 1 above)
Comment (robertbastian) [reply utils/zerotrie/examples/first_weekday_for_region.rs:218]: Yeah, utils use `assert_eq!` in a lot of places. It means the binsize benches won't be accurate, but we can address this later.

**Review by sffc — APPROVED** [review]

File: tools/benchmark/macros/src/lib.rs [hunk 2]
```diff
@@ -16,64 +16,98 @@
...
-    #[macro_export]
-    macro_rules! main_setup {
-        () => {};
-    }
+#[macro_export]
+macro_rules! instrument {
+    () => {
+        const _: () = {
+            #[no_mangle]
+            fn main(_argc: isize, _argv: *const *const u8) -> isize {
```
Comment (Manishearth) [inline tools/benchmark/macros/src/lib.rs:64]: suggestion: we can also just call it `actual_main` and give it a `link_name = main` attribute

**Review by Manishearth — APPROVED** [review]

File: tools/benchmark/macros/src/lib.rs (hunk 2 above)
Comment (robertbastian) [reply tools/benchmark/macros/src/lib.rs:64]: I tried this, it doesn't seem to work.

**Pull request merged**